test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -156,6 +158,318 @@ public void UpdatePackageFromUI()
+            CommonUtility.AssertPackageInPackagesConfig(VisualStudio, nuProject, "contoso.a", "1.0.0", XunitLogger);
+        }
+
+        [StaFact]
+        public async Task InstallPackageToProjectsFromUI_PackageNamespace_WithSingleFeed_NoMatch_Succeeds()
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+            var solutionService = VisualStudio.Get<SolutionService>();
+            string solutionDirectory = CommonUtility.CreateSolutionDirectory(Directory.GetCurrentDirectory());
+            solutionService.CreateEmptySolution("TestSolution", solutionDirectory);
+
+            var privateRepositoryPath = Path.Combine(solutionService.ContainingDirectory, "PrivateRepository");
+            Directory.CreateDirectory(privateRepositoryPath);
+
+            var packageName = "Contoso.A";
+            var packageVersion = "1.0.0";
+
+            await CommonUtility.CreatePackageInSourceAsync(privateRepositoryPath, packageName, packageVersion);
+
+            //Create nuget.config with Package namespace filtering rules.
+            CommonUtility.CreateConfigurationFile(Path.Combine(solutionDirectory, "NuGet.config"), $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+    <add key=""PrivateRepository"" value=""{privateRepositoryPath}"" />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key=""PrivateRepository"">
+            <namespace id=""ContosoA.*"" /> <!-- Filter doesn't match Contoso.A -->
+            <namespace id=""Test.*"" />
+        </packageSource>
+    </packageNamespaces>
+//</configuration>");
+
+            var project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.ClassLibrary, ProjectTargetFramework.V46, "TestProject");
+            var nuProject = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.ClassLibrary, ProjectTargetFramework.V46, "NuProject");
+            solutionService.SaveAll();
+
+            // Act
+            OpenNuGetPackageManagerWithDte();
+            var nugetTestService = GetNuGetTestService();
+            var uiwindow = nugetTestService.GetUIWindowfromProject(nuProject);
+            uiwindow.InstallPackageFromUI("contoso.a", "1.0.0");
+            VisualStudio.SelectProjectInSolutionExplorer(project.Name);
+            OpenNuGetPackageManagerWithDte();
+
+            VisualStudio.ClearOutputWindow();
+            var uiwindow2 = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow2.InstallPackageFromUI("contoso.a", "1.0.0");
+
+            // Assert
+            CommonUtility.AssertPackageInPackagesConfig(VisualStudio, project, "contoso.a", "1.0.0", XunitLogger);
+            CommonUtility.AssertPackageInPackagesConfig(VisualStudio, nuProject, "contoso.a", "1.0.0", XunitLogger);
+        }
+
+        [StaFact]
```

Comment:
@nkolev92 
Here I was not able to add `InstallPackageFromUI_PackageNamespace_WithMultiFeed_Success()` test due to here unable to change source option to `All` with current Apex contract. Any change requires not only change in Apex contract also additional change (other to Apex test) in NuGet PMUI code is kind of out of scope of this PR.
Previously I thought NuGet PMUI searches from all sources for installing new package but it seems it's not true, if individual source is selected then it searches only from that source. It doesn't consider `Mypackages` for example in this case.
![image](https://user-images.githubusercontent.com/8766776/125343299-c3dab300-e30a-11eb-8734-dc47ebcb6801.png)
In order to make multifeed to work I need to select `All`, but currently not available due to previous mentioned limitation.

Reply:
> . Any change requires not only change in Apex contract also additional change (other to Apex test) in NuGet PMUI code is kind of out of scope of this PR.

Not sure I agree here. 

We should add a test for this scenario. Apex test contracts are owned by us, it's equivalent to making a change to a type like: https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/Commands/TestRestoreRequest.cs

You can argue that this can be added in a new PR, but it's something that I'd say we should consider as part of the 6.0 MVP.

Reply:
It can be done but changes would be kind of ugly, I have to change private methods/fields to public.
It only benefits Apex test but nothing else, so it would be to hard to justify it.

Reply:
@nkolev92 
Can I create follow up issue for this Apex test case and do it separate PR?

Reply:
> It can be done but changes would be kind of ugly, I have to change private methods/fields to public.
> It only benefits Apex test but nothing else, so it would be to hard to justify it.

Code in NuGet.Clients can be refactored as needed. 
Making the code testable would be a good investment. 

Do you expect this to be a very involved work?

> Can I create follow up issue for this Apex test case and do it separate PR?

Yep, I'm comfortable with that. In the meantime, please ensure the scenarios work manually (I know you've done that :) )

Reply:
Ok. I created follow up issue https://github.com/NuGet/Home/issues/11026
Until this one is completed below 2 new unit tests technically cover it.

https://github.com/NuGet/NuGet.Client/blob/a8975bb6651dfbb4c50781895a2b48833304dcc8/test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs#L7117-L7121

https://github.com/NuGet/NuGet.Client/blob/a8975bb6651dfbb4c50781895a2b48833304dcc8/test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs#L7219-L7222